src/notifications/reducer.ts

```diff
@@ -0,0 +1,84 @@
+import { createReducer } from 'redux-act'
+
+import * as actions from './actions'
+
+export interface State {
+    readNotificationList: {
+        notifications: any[]
+        resultExhausted: boolean
+    }
+    unreadNotificationList: any[]
+    showMoreIndex?: any
+    unreadNotifications: number
+    currentPage: number
+    isLoading: boolean
+    isReadExpanded: boolean
+    showInbox: boolean
+    unreadNotifCount: number
+    shouldTrack: boolean
+}
+
+const defaultState: State = {
+    readNotificationList: {
+        notifications: [],
+        resultExhausted: false,
+    },
+    unreadNotificationList: [],
+    showMoreIndex: undefined,
+    unreadNotifications: 0,
```

Comment:
We are using the `unreadNotifications` in the overview container and popup container, we are just using the selector of the notifications. Therefore no need to remove. :slightly_smiling_face:

Reply:
`unreadNotifications` (`number` type) is not being used in either container; there is not even a corresponding selector or reducer for it (you may be confusing it with `unreadNotificationList`? -`any[]` type). That should be removed.

However, the most important part of my point here is to unify the `unreadNotificationList` and `readNotificationList.notifications` state (unless you can think of a good reason not to)

Reply:
Here we cannot unify because we have to show these things in the different sections if we will do that we have to apply the filter on the UI side for the `unreadNotificationList` and `readNotificationList`, that makes odd.
Make sense?

Reply:
Nothing needs to change in your containers+views part of the UI. The only thing you need to change to unify this is here in the related reducers and in the related selectors. The container should receive the exact same data as it does now.